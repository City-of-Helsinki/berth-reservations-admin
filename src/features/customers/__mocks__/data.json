{
  "profiles": {
    "__typename": "ProfileNodeConnection",
    "edges": [
      {
        "__typename": "ProfileNodeEdge",
        "node": {
          "__typename": "ProfileNode",
          "id": "UHJvZmlsZVR5cGU6YzgwMWM4MGUtOGZiYS00N2I4LWE1YjUtYWE4ZGY5YmVlOTEy",
          "firstName": "Jussi",
          "lastName": "Virtanen",
          "nickname": "virtajussi",
          "primaryAddress": {
            "address": "Virtasentie 1",
            "city": "Helsinki",
            "postalCode": "00100"
          },
          "primaryPhone": null,
          "primaryEmail": {
            "email": "test@example.com"
          },
          "serviceConnections": null,
          "contactMethod": "EMAIL",
          "image": ""
        }
      },
      {
        "__typename": "ProfileNodeEdge",
        "node": {
          "__typename": "ProfileNode",
          "id": "UHJvZmlsZVR5cGU6YzgwMWM4MGUtOGZiYS00N2I4LWE1YjUtYWE4ZGY5YmVlOTE",
          "firstName": "Jussi2",
          "lastName": "Virtanen",
          "nickname": "virtajussi",
          "organization": {
            "businessId": "1234567-8",
            "organizationType": "COMPANY"
          },
          "primaryAddress": {
            "address": "Virtasentie 1",
            "city": "Helsinki",
            "postalCode": "00100"
          },
          "primaryPhone": null,
          "primaryEmail": {
            "email": "test@example.com"
          },
          "serviceConnections": null,
          "contactMethod": "EMAIL",
          "image": ""
        }
      }
    ]
  }
}