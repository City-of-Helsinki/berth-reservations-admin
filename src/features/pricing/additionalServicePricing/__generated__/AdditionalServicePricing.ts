/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { ProductServiceType, PriceUnits, PeriodType, AdditionalProductType, AdditionalProductTaxEnum } from "./../../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL fragment: AdditionalServicePricing
// ====================================================

export interface AdditionalServicePricing_edges_node {
  __typename: "AdditionalProductNode";
  id: string;
  service: ProductServiceType;
  priceValue: any;
  priceUnit: PriceUnits;
  period: PeriodType;
  productType: AdditionalProductType;
  taxPercentage: AdditionalProductTaxEnum;
}

export interface AdditionalServicePricing_edges {
  __typename: "AdditionalProductNodeEdge";
  node: AdditionalServicePricing_edges_node | null;
}

export interface AdditionalServicePricing {
  __typename: "AdditionalProductNodeConnection";
  edges: (AdditionalServicePricing_edges | null)[];
}
