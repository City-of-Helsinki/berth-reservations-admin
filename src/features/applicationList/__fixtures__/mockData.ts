import { BERTH_APPLICATIONS } from '../__generated__/BERTH_APPLICATIONS';
import { getBerthApplicationData } from '../utils';
import { ApplicationStatus, LeaseStatus } from '../../../@types/__generated__/globalTypes';

export const mockData: BERTH_APPLICATIONS = {
  berthApplications: {
    __typename: 'BerthApplicationNodeConnection',
    count: 3,
    edges: [
      // Full structure
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: {
            __typename: 'BerthSwitchType',
            berthNumber: '1',
            harbor: 'MOCK-HARBOR-0',
            harborName: 'Testisatama',
            id: 'MOCK-BERTH-SWITCH-0',
            pier: 'A',
            reason: { __typename: 'BerthSwitchReasonType', title: 'Reason' },
          },
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano',
          boatDraught: 5,
          boatRegistrationNumber: 'A 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE' },
          firstName: 'Matti',
          harborChoices: [{ __typename: 'HarborChoiceType', harbor: '1', harborName: '1', priority: 0 }],
          id: 'MOCK-APPLICATION-0',
          lastName: 'Meikäläinen',
          lease: {
            __typename: 'BerthLeaseNode',
            berth: {
              __typename: 'BerthNode',
              number: '7',
              pier: {
                __typename: 'PierNode',
                properties: {
                  __typename: 'PierProperties',
                  harbor: {
                    __typename: 'HarborNode',
                    id: 'MOCK-HARBOR-0',
                    properties: { __typename: 'HarborProperties', name: 'Testisatama' },
                  },
                  identifier: 'G',
                },
              },
            },
            id: 'MOCK-LEASE-0',
            status: LeaseStatus.OFFERED,
          },
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
      // Most structure and nulled fields
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: {
            __typename: 'BerthSwitchType',
            berthNumber: '2',
            harbor: 'MOCK-HARBOR-0',
            harborName: 'Testisatama',
            id: 'MOCK-BERTH-SWITCH-1',
            pier: 'A',
            reason: null,
          },
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano II',
          boatDraught: 5,
          boatRegistrationNumber: 'B 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE' },
          firstName: 'Maija',
          harborChoices: null,
          id: 'MOCK-APPLICATION-1',
          lastName: 'Meikäläinen',
          lease: {
            __typename: 'BerthLeaseNode',
            id: 'MOCK-LEASE-1',
            status: LeaseStatus.OFFERED,
            berth: {
              __typename: 'BerthNode',
              number: '7',
              pier: {
                __typename: 'PierNode',
                properties: {
                  __typename: 'PierProperties',
                  identifier: 'Pier Brosnan',
                  harbor: {
                    __typename: 'HarborNode',
                    id: 'MOCK-HARBOR-1',
                    properties: {
                      __typename: 'HarborProperties',
                      name: null,
                    },
                  },
                },
              },
            },
          },
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
      // Minimum fields
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: null,
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano 3-D',
          boatDraught: 5,
          boatRegistrationNumber: 'C 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE-0' },
          firstName: 'Matti',
          harborChoices: [
            { __typename: 'HarborChoiceType', harbor: 'MOCK-HARBOR-0', harborName: 'Testisatama', priority: 0 },
          ],
          id: 'MOCK-APPLICATION-2',
          lastName: 'Möttönen',
          lease: null,
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
    ],
  },
  boatTypes: [{ __typename: 'BoatTypeType', id: '1', name: 'Purjevene' }],
};

export const mockTableData = getBerthApplicationData(mockData);
