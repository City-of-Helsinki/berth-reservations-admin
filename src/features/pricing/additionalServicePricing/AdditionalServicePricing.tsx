import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Row } from 'react-table';
import { useMutation } from '@apollo/react-hooks';

import Card from '../../../common/card/Card';
import CardHeader from '../../../common/cardHeader/CardHeader';
import CardBody from '../../../common/cardBody/CardBody';
import Table, { Column, COLUMN_WIDTH } from '../../../common/table/Table';
import Text from '../../../common/text/Text';
import EditForm, { EDIT_FORM_TYPE } from '../editModal/EditForm';
import { AdditionalServicePricing as AdditionalServicePricingData } from './__generated__/AdditionalServicePricing';
import { getAdditionalServiceData } from './utils';
import { getPeriodTKey, getProductServiceTKey, getProductTax } from '../../../common/utils/translations';
import { PeriodType, ProductServiceType, AdditionalProductTaxEnum } from '../../../@types/__generated__/globalTypes';
import EditModal from '../editModal/EditModal';
import { UPDATE_ADDITIONAL_SERVICE_PRICE_MUTATION } from './mutations';
import { UPDATE_ADDITIONAL_SERVICE_PRICE } from './__generated__/UPDATE_ADDITIONAL_SERVICE_PRICE';
import { UPDATE_HARBOR_SERVICE_PRICEVariables as UPDATE_HARBOR_SERVICE_PRICE_VARS } from '../harborServicePricing/__generated__/UPDATE_HARBOR_SERVICE_PRICE';

export interface AdditionalService {
  id: string;
  service: ProductServiceType;
  price: string;
  tax: AdditionalProductTaxEnum;
  period: PeriodType;
}

export interface AdditionalServicePricingProps {
  data: AdditionalServicePricingData | undefined | null;
  loading: boolean;
  className?: string;
}

const AdditionalServicePricing = ({ data, loading, className }: AdditionalServicePricingProps) => {
  const { t, i18n } = useTranslation();

  const [editRowValues, setEditRowValues] = useState<AdditionalService>();
  const [updateHarborServicePrice] = useMutation<UPDATE_ADDITIONAL_SERVICE_PRICE, UPDATE_HARBOR_SERVICE_PRICE_VARS>(
    UPDATE_ADDITIONAL_SERVICE_PRICE_MUTATION
  );

  const additionalServicesCols: Column<AdditionalService>[] = [
    {
      Header: t('pricing.additionalServices.service') || '',
      width: COLUMN_WIDTH.L,
      accessor: 'service',
      Cell: ({ cell }) => t(getProductServiceTKey(cell.value)),
    },
    {
      Header: t('pricing.additionalServices.price') || '',
      width: COLUMN_WIDTH.XS,
      accessor: 'price',
    },
    {
      Header: t('pricing.additionalServices.tax') || '',
      width: COLUMN_WIDTH.XS,
      accessor: 'tax',
      Cell: ({ cell }) => getProductTax(cell.value, i18n.language),
    },
    {
      Header: t('pricing.additionalServices.period') || '',
      width: COLUMN_WIDTH.S,
      accessor: 'period',
      Cell: ({ cell }) => t(getPeriodTKey(cell.value)),
    },
    {
      id: 'edit',
      Header: t('common.edit') || '',
      sortType: 'none',
      width: COLUMN_WIDTH.S,
      Cell: ({ row }: { row: Row<AdditionalService> }) => (
        <button onClick={() => setEditRowValues(row.original)}>
          <Text color="brand">{t('common.edit')}</Text>
        </button>
      ),
    },
  ];

  const handleSubmit = async ({ id, price, period, tax }: AdditionalService) => {
    await updateHarborServicePrice({
      variables: { input: { id, priceValue: price, period, taxPercentage: tax } },
    });

    setEditRowValues(undefined);
  };

  const handleClose = () => setEditRowValues(undefined);

  return (
    <>
      <Card className={className}>
        <CardHeader title={t('pricing.additionalServices.title')} />
        <CardBody>
          <Table
            columns={additionalServicesCols}
            data={getAdditionalServiceData(data)}
            loading={loading}
            theme="basic"
            renderEmptyStateRow={() => t('common.notification.noData.description')}
          />
        </CardBody>
      </Card>
      <EditModal isOpen={!!editRowValues} closeModal={handleClose}>
        {editRowValues && (
          <EditForm
            closeModal={handleClose}
            formType={EDIT_FORM_TYPE.ADDITIONAL_SERVICES}
            initialValues={editRowValues}
            onSubmit={handleSubmit}
          />
        )}
      </EditModal>
    </>
  );
};

export default AdditionalServicePricing;
