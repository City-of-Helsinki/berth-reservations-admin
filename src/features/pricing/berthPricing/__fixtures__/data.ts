import { PlaceProductTaxEnum, PriceUnits, PricingCategory } from '../../../../@types/__generated__/globalTypes';
import { BerthPricingProps } from '../BerthPricing';

export const data: BerthPricingProps['data'] = {
  __typename: 'BerthProductNodeConnection',
  edges: [
    {
      __typename: 'BerthProductNodeEdge',
      node: {
        __typename: 'BerthProductNode',
        id: '68ac7db5-b397-48f6-af19-99c35d645cb6',
        minWidth: 3,
        maxWidth: 7,
        tier1Price: 50,
        tier2Price: 75,
        tier3Price: 100,
        priceUnit: PriceUnits.AMOUNT,
        taxPercentage: PlaceProductTaxEnum.TAX_24_00,
        pricingCategory: PricingCategory.DEFAULT,
      },
    },
    {
      __typename: 'BerthProductNodeEdge',
      node: {
        __typename: 'BerthProductNode',
        id: '82c3ba8c-f409-4f54-ad8f-12ace3f5df50',
        minWidth: 6,
        maxWidth: 14,
        tier1Price: 150,
        tier2Price: 175,
        tier3Price: 200,
        priceUnit: PriceUnits.AMOUNT,
        taxPercentage: PlaceProductTaxEnum.TAX_24_00,
        pricingCategory: PricingCategory.DEFAULT,
      },
    },
    {
      __typename: 'BerthProductNodeEdge',
      node: {
        __typename: 'BerthProductNode',
        id: '27a112bd-734c-461c-86d1-ba78d7351a3f',
        minWidth: 3,
        maxWidth: 7,
        tier1Price: 40,
        tier2Price: 60,
        tier3Price: 80,
        priceUnit: PriceUnits.AMOUNT,
        taxPercentage: PlaceProductTaxEnum.TAX_24_00,
        pricingCategory: PricingCategory.VASIKKASAARI,
      },
    },
  ],
};
