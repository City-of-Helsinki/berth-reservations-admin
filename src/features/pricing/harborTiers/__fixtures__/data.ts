import { PriceTier } from '../../../../@types/__generated__/globalTypes';
import { HarborTiersProps } from '../HarborTiers';

export const data: HarborTiersProps['data'] = {
  __typename: 'HarborNodeConnection',
  edges: [
    {
      __typename: 'HarborNodeEdge',
      node: {
        __typename: 'HarborNode',
        id: '98b7ab56-9c67-46dd-81b2-9eb6f66cbabe',
        properties: {
          __typename: 'HarborProperties',
          name: 'Pound Sterling',
          piers: {
            __typename: 'PierNodeConnection',
            edges: [
              {
                __typename: 'PierNodeEdge',
                node: {
                  __typename: 'PierNode',
                  id: '2030fa0e-c125-44dd-b44e-2b9be403cff6',
                  properties: {
                    __typename: 'PierProperties',
                    priceTier: PriceTier.TIER_1,
                  },
                },
              },
            ],
          },
        },
      },
    },
  ],
};
