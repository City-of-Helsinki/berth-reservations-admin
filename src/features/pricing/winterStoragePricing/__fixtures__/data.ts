import { WinterStoragePricingProps } from '../WinterStoragePricing';
import { PriceUnits } from '../../../../@types/__generated__/globalTypes';

export const data: WinterStoragePricingProps['data'] = {
  __typename: 'WinterStorageAreaNodeConnection',
  edges: [
    {
      __typename: 'WinterStorageAreaNodeEdge',
      node: {
        __typename: 'WinterStorageAreaNode',
        id: 'b3cb95f7-e9ce-438a-8447-203145f047fe',
        properties: {
          __typename: 'WinterStorageAreaProperties',
          name: 'Corporate Developer Mobility',
          product: {
            __typename: 'WinterStorageProductNode',
            priceValue: '10.2',
            priceUnit: PriceUnits.AMOUNT,
          },
        },
      },
    },
  ],
};
