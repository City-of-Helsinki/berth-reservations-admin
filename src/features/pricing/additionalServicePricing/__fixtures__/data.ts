import { AdditionalServicePricingProps } from '../AdditionalServicePricing';
import {
  ProductServiceType,
  PriceUnits,
  PeriodType,
  AdditionalProductType,
  AdditionalProductTaxEnum,
} from '../../../../@types/__generated__/globalTypes';

export const data: AdditionalServicePricingProps['data'] = {
  __typename: 'AdditionalProductNodeConnection',
  edges: [
    {
      __typename: 'AdditionalProductNodeEdge',
      node: {
        __typename: 'AdditionalProductNode',
        id: 'a8400b59-534c-4934-b643-083a4273ca1a',
        service: ProductServiceType.DINGHY_PLACE,
        priceValue: 10,
        priceUnit: PriceUnits.AMOUNT,
        period: PeriodType.SEASON,
        productType: AdditionalProductType.OPTIONAL_SERVICE,
        taxPercentage: AdditionalProductTaxEnum.TAX_24_00,
      },
    },
  ],
};
