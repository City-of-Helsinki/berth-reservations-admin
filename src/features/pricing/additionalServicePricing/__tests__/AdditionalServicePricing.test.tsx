import React from 'react';
import { shallow } from 'enzyme';

import AdditionalServicePricing, { AdditionalServicePricingProps } from '../AdditionalServicePricing';
import { data } from '../__fixtures__/data';

describe('AdditionalServicePricing', () => {
  const initialProps: AdditionalServicePricingProps = { data, loading: false };

  const getWrapper = (props: Partial<AdditionalServicePricingProps> = {}) =>
    shallow(<AdditionalServicePricing {...initialProps} {...props} />);

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders normally', () => {
    const wrapper = getWrapper();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders noData message when there is no data', () => {
    const wrapper = getWrapper({ data: undefined });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
