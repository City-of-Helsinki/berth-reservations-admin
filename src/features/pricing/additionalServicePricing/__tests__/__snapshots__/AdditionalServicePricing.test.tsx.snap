// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdditionalServicePricing renders noData message when there is no data 1`] = `"<div class=\\"card\\"><div class=\\"cardHeader\\"><span class=\\"title\\">Lisäpalvelut</span></div><div class=\\"cardBody\\"><div><div class=\\"tableWrapper\\"><div role=\\"table\\" style=\\"min-width: 600px;\\" class=\\"table basic noMainHeader\\"><div class=\\"stickyHeaders\\"><div role=\\"row\\" style=\\"display: flex; flex: 1 0 auto; min-width: 0px;\\" class=\\"header\\"><div colspan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing: border-box; flex: 225 0 auto; min-width: 0px; width: 225px; cursor: pointer;\\" title=\\"Toggle SortBy\\" class=\\"headerCell\\">Palvelu<div class=\\"sortArrowWrapper\\"><svg class=\\"Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_m__3edUY icon_hds-icon--size-m__1mcHv sortArrow sortArrowDown\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\"><g fill=\\"none\\" fill-rule=\\"evenodd\\"><path d=\\"M0 0h24v24H0z\\"></path><path fill=\\"currentColor\\" d=\\"M10 5.636l1.414 1.414L7.465 11H20v2H7.464l3.95 3.95L10 18.364 3.636 12z\\"></path></g></svg></div></div><div colspan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing: border-box; flex: 75 0 auto; min-width: 0px; width: 75px; cursor: pointer;\\" title=\\"Toggle SortBy\\" class=\\"headerCell\\">Hinta<div class=\\"sortArrowWrapper\\"><svg class=\\"Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_m__3edUY icon_hds-icon--size-m__1mcHv sortArrow sortArrowDown\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\"><g fill=\\"none\\" fill-rule=\\"evenodd\\"><path d=\\"M0 0h24v24H0z\\"></path><path fill=\\"currentColor\\" d=\\"M10 5.636l1.414 1.414L7.465 11H20v2H7.464l3.95 3.95L10 18.364 3.636 12z\\"></path></g></svg></div></div><div colspan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing: border-box; flex: 75 0 auto; min-width: 0px; width: 75px; cursor: pointer;\\" title=\\"Toggle SortBy\\" class=\\"headerCell\\">ALV<div class=\\"sortArrowWrapper\\"><svg class=\\"Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_m__3edUY icon_hds-icon--size-m__1mcHv sortArrow sortArrowDown\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\"><g fill=\\"none\\" fill-rule=\\"evenodd\\"><path d=\\"M0 0h24v24H0z\\"></path><path fill=\\"currentColor\\" d=\\"M10 5.636l1.414 1.414L7.465 11H20v2H7.464l3.95 3.95L10 18.364 3.636 12z\\"></path></g></svg></div></div><div colspan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing: border-box; flex: 112.5 0 auto; min-width: 0px; width: 112.5px; cursor: pointer;\\" title=\\"Toggle SortBy\\" class=\\"headerCell\\">Aika<div class=\\"sortArrowWrapper\\"><svg class=\\"Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_m__3edUY icon_hds-icon--size-m__1mcHv sortArrow sortArrowDown\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\"><g fill=\\"none\\" fill-rule=\\"evenodd\\"><path d=\\"M0 0h24v24H0z\\"></path><path fill=\\"currentColor\\" d=\\"M10 5.636l1.414 1.414L7.465 11H20v2H7.464l3.95 3.95L10 18.364 3.636 12z\\"></path></g></svg></div></div><div colspan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing: border-box; flex: 112.5 0 auto; min-width: 0px; width: 112.5px;\\" class=\\"headerCell\\">Muokkaa</div></div></div><div role=\\"rowgroup\\"><div class=\\"rowWrapper\\" role=\\"row\\"><div class=\\"tableCell\\">Ei tietoja</div></div></div></div></div></div></div> </div>"`;

exports[`AdditionalServicePricing renders normally 1`] = `"<div class=\\"card\\"><div class=\\"cardHeader\\"><span class=\\"title\\">Lisäpalvelut</span></div><div class=\\"cardBody\\"><div><div class=\\"tableWrapper\\"><div role=\\"table\\" style=\\"min-width: 600px;\\" class=\\"table basic noMainHeader\\"><div class=\\"stickyHeaders\\"><div role=\\"row\\" style=\\"display: flex; flex: 1 0 auto; min-width: 0px;\\" class=\\"header\\"><div colspan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing: border-box; flex: 225 0 auto; min-width: 0px; width: 225px; cursor: pointer;\\" title=\\"Toggle SortBy\\" class=\\"headerCell\\">Palvelu<div class=\\"sortArrowWrapper\\"><svg class=\\"Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_m__3edUY icon_hds-icon--size-m__1mcHv sortArrow sortArrowDown\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\"><g fill=\\"none\\" fill-rule=\\"evenodd\\"><path d=\\"M0 0h24v24H0z\\"></path><path fill=\\"currentColor\\" d=\\"M10 5.636l1.414 1.414L7.465 11H20v2H7.464l3.95 3.95L10 18.364 3.636 12z\\"></path></g></svg></div></div><div colspan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing: border-box; flex: 75 0 auto; min-width: 0px; width: 75px; cursor: pointer;\\" title=\\"Toggle SortBy\\" class=\\"headerCell\\">Hinta<div class=\\"sortArrowWrapper\\"><svg class=\\"Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_m__3edUY icon_hds-icon--size-m__1mcHv sortArrow sortArrowDown\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\"><g fill=\\"none\\" fill-rule=\\"evenodd\\"><path d=\\"M0 0h24v24H0z\\"></path><path fill=\\"currentColor\\" d=\\"M10 5.636l1.414 1.414L7.465 11H20v2H7.464l3.95 3.95L10 18.364 3.636 12z\\"></path></g></svg></div></div><div colspan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing: border-box; flex: 75 0 auto; min-width: 0px; width: 75px; cursor: pointer;\\" title=\\"Toggle SortBy\\" class=\\"headerCell\\">ALV<div class=\\"sortArrowWrapper\\"><svg class=\\"Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_m__3edUY icon_hds-icon--size-m__1mcHv sortArrow sortArrowDown\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\"><g fill=\\"none\\" fill-rule=\\"evenodd\\"><path d=\\"M0 0h24v24H0z\\"></path><path fill=\\"currentColor\\" d=\\"M10 5.636l1.414 1.414L7.465 11H20v2H7.464l3.95 3.95L10 18.364 3.636 12z\\"></path></g></svg></div></div><div colspan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing: border-box; flex: 112.5 0 auto; min-width: 0px; width: 112.5px; cursor: pointer;\\" title=\\"Toggle SortBy\\" class=\\"headerCell\\">Aika<div class=\\"sortArrowWrapper\\"><svg class=\\"Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_m__3edUY icon_hds-icon--size-m__1mcHv sortArrow sortArrowDown\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\"><g fill=\\"none\\" fill-rule=\\"evenodd\\"><path d=\\"M0 0h24v24H0z\\"></path><path fill=\\"currentColor\\" d=\\"M10 5.636l1.414 1.414L7.465 11H20v2H7.464l3.95 3.95L10 18.364 3.636 12z\\"></path></g></svg></div></div><div colspan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing: border-box; flex: 112.5 0 auto; min-width: 0px; width: 112.5px;\\" class=\\"headerCell\\">Muokkaa</div></div></div><div role=\\"rowgroup\\"><div class=\\"rowWrapper\\"><div role=\\"row\\" style=\\"display: flex; flex: 1 0 auto; min-width: 0px;\\"><div role=\\"cell\\" style=\\"box-sizing: border-box; flex: 225 0 auto; min-width: 0px; width: 225px;\\" class=\\"tableCell\\">Jollapaikka</div><div role=\\"cell\\" style=\\"box-sizing: border-box; flex: 75 0 auto; min-width: 0px; width: 75px;\\" class=\\"tableCell\\">10,00&nbsp;€</div><div role=\\"cell\\" style=\\"box-sizing: border-box; flex: 75 0 auto; min-width: 0px; width: 75px;\\" class=\\"tableCell\\">24&nbsp;%</div><div role=\\"cell\\" style=\\"box-sizing: border-box; flex: 112.5 0 auto; min-width: 0px; width: 112.5px;\\" class=\\"tableCell\\">Kausi</div><div role=\\"cell\\" style=\\"box-sizing: border-box; flex: 112.5 0 auto; min-width: 0px; width: 112.5px;\\" class=\\"tableCell\\"><button><span class=\\"text brand span\\">Muokkaa</span></button></div></div></div></div></div></div></div></div> </div>"`;
